src/lib.rs

```diff
@@ -626,6 +626,105 @@ where
+                    ),
+                },
+            },
+        };
+        while (next_cluster.0 - cluster.0) == 1 {
+            cluster = next_cluster;
+            next_cluster = match &volume.volume_type {
+                VolumeType::Fat(fat) => match fat.next_cluster(self, cluster) {
+                    Ok(cluster) => cluster,
+                    Err(e) => match e {
+                        Error::EndOfFile => break,
+                        _ => panic!(
+                            "Error: traversing the FAT table, accessed free space or a bad cluster"
+                        ),
+                    },
+                },
+            };
+            contiguous_cluster_count += 1;
+        }
+        Ok(contiguous_cluster_count)
+    }
+
+    /// Read from an open file. It has the same effect as the [`Self::read`] method but reduces `read time`
+    /// by more than 50%, especially in the case of large files (i.e. > 1Mb)
+    ///
+    /// `read_multi` reads multiple contiguous blocks of a file in a single read operation,
+    /// without the extra overhead of additional `data-copying`.
+    ///
+    /// NOTE: The buffer argument must be a multiple of 512 bytes. This impl assumes the underlying
+    /// emmc driver (and consequently the EMMC device) features support for multi-block reads.
+    #[cfg(feature = "unstable")]
+    pub fn read_multi(
+        &mut self,
+        volume: &Volume,
+        file: &mut File,
+        buffer: &mut [u8],
+    ) -> Result<usize, Error<D::Error>> {
+        let blocks_per_cluster = match &volume.volume_type {
+            VolumeType::Fat(fat) => fat.blocks_per_cluster,
+        };
+
+        let mut bytes_read = 0;
+        let mut block_read_counter = 0;
+        let mut starting_cluster = file.starting_cluster;
+        let mut file_blocks;
+        if (file.length % Block::LEN as u32) == 0 {
+            file_blocks = file.length / Block::LEN as u32;
+        } else {
+            file_blocks = (file.length / Block::LEN as u32) + 1;
+        }
+
+        while file_blocks > 0 {
+            // Walk the FAT to see if we have contiguos clusters
+            let contiguous_cluster_count =
+                self.check_contiguous_cluster_count(volume, starting_cluster)?;
+
+            let blocks_to_read = contiguous_cluster_count * blocks_per_cluster as u32;
+            let bytes_to_read = Block::LEN * blocks_to_read as usize;
+            let (blocks, _) = buffer[block_read_counter..block_read_counter + bytes_to_read]
+                .as_chunks_mut::<{ Block::LEN }>();
+            // `cluster_to_block` gives us the absolute block_idx i.e. gives us the block offset from the 0th Block
+            let block_idx = match &volume.volume_type {
+                VolumeType::Fat(fat) => fat.cluster_to_block(starting_cluster),
+            };
+
+            self.block_device
+                .read(Block::from_array_slice(blocks), block_idx, "read")
```

Comment:
The string argument I *think* should be the function name (but it's been a long while since I wrote this so I might be wrong)

Reply:
Changed from ~~read~~ to `read_multi`